### DISCUSSION TITLE
不具合報告はこちら

### DISCUSSION THREAD

**Discussion by dongaba** [dongaba/TVerRec#14] [Issues]
不具合報告はこちらにお願いします。
起きている不具合を具体的に説明してください。
また、使用しているOS、バージョンなどの情報も記載ください。
特定の動画で起きている問題の場合は、その動画のURLなどもお願いします。

**Comment by BRABUS-RC** [comment]
dongabaさん　こんばんは。
先程よりDLしておりますが、「自動」も「個別」もDL出来なくなってます。
dongabaさんの方は問題無いでしょうか。

**Reply by dongaba** [reply]
私の環境でもDLできなくなっていました。
使用しているyoutube-dlのツールが原因のようでしたので、別のyoutube-dlを取得するように差し替えました。
v2.4.8にバージョンアップして試してみてください。

**Comment by BRABUS-RC** [comment]
dongabaさん、こんばんは。
今、起動確認致しました。
問題無くDL出来ました。バージョンアップありがとうございました。

**Reply by dongaba** [reply]
問題が解決したようで良かったです

**Comment by royal-bk** [comment]
dongabaさん、はじめまして。
素晴らしいソフトをありがとうございます。

「町中華で飲ろうぜ」という番組で２つエピソードが配信されてたのですが、２つとも同じファイル名になるエピソードだったらしく、ファイルが１つだけになってました。

町中華で飲ろうぜ 2022年 2022年04月19日放送 新潟SP
https://tver.jp/episodes/ep5ko7xuct
https://tver.jp/episodes/epgivhgk3i
(２つとも、すでにリンク切れ)

また、調べたところ…

"番組名 放送年 放送日 長いサブタイトル（前編）"
"番組名 放送年 放送日 長いサブタイトル（後編）"

のような番組のエピソードがある場合、SMB対策で行われるファイル名のトリミングにより

"番組名 放送年 放送日 長いサブタイト…"

となり、同上の現象が発生する可能性があるようです。

現状、ほとんど遭遇しない問題ですが一応報告しておきます。

使用OS・バージョンは Windows 10 Pro バージョン 21H2 です。

**Reply by dongaba** [reply]
なるほど、ご報告をありがとうございます。
残念ながら町中華で飲ろうぜは私が録画している番組ではなく、既にLinkも切れてしまっているので検証できず。。
ただ、バラエティー番組を中心にサブタイトルが長い番組が多いのは確かなので、他の番組でも起こりうるかもしれないですね。

一方で、トリミングしないと保存エラーが多発してしまいますし、サブタイトルの前半や途中をトリミングするわけにもいかないので悩ましいところです。
トリミング後の名前が重複していないかチェックして連番を振ることも可能ではありますが、ちょっと今すぐ直せそうにはありません。

現時点でご自分でやっていただける対策としては、Cドライブ直下にフォルダを作るなどして、そこを保存先にしてできるだけファイル名のトリミングがされないようにしていただくことくらいしか思いつきません。

**Reply by dongaba** [reply]
時期を含め解決を保証することはできませんが、IssueとしてLogしておきました。
https://github.com/dongaba/TVerRec/issues/16

**Reply by dongaba** [reply]
まだ正式リリースしておりませんが、v2.6.6にて対応しました。

**Comment by BRABUS-RC** [comment]
dongabaさん、こんばんは。
お世話になっております。
バージョン2.5.3bata2更新ありがとうございます。
2.5.3bata2で作動がおかしくなったので調べて頂けませんか。
conf→keyword.sample.conf（新しくリスト作成）で自動ダウンロードに於いて「ダウンロード対象ジャンルリストが存在しません。終了します」と赤文字表示され（他にも英語で色々表示）最後に「終了するには Y と入力してください。何も入力しなければ処理を継続します。3600秒待機します... [Y,N]?」となります。
個別ダウンロードでも「ダウンロード対象ジャンルリストが存在しません。終了します」と赤文字表示され最後に「Press any key to continue . . .」となります。


conf→keyword.conf（バージョンアップする度に最初に作ったkeywordコピーして使ってる）に変更すると、個別ダウンロードで「TVerから情報を取得できませんでした。スキップします」となります。
自動ダウンロードでは、「TVerから情報を取得できませんでした。スキップします」と表示する物や「正常にダウンロードする物」もあります。

ちなみにバージョン2.5.0でconf→keyword.confでは自動、個別共に正常にダウンロード出来てます。

説明が下手ですみませんが宜しくお願い致します。

**Reply by dongaba** [reply]
おそらくですが、KeywordファイルとIgnoreファイルの名前が間違っているのではないでしょうか?
Keywordファイルについては、keyword.sample.confという名前で保存するのではなく、keyword.confという名前で保存する必要があります。
Ignoreファイルについても同様で、sampleを含まないファイル名で保存しておいてください。
なお、Keywordファイルについて、バージョンアップの際に過去のKeywordファイルをコピーして使っていただいているようですが、v2.5.0よりジャンルの指定方法が若干変わっているのでご注意ください。

**Reply by BRABUS-RC** [reply]
こんばんは。
ソフトダウンロード後解凍したら「keyword.sample.conf」となっていたので名前が変更になってると思ってました。　昼に書き換えてテストしてみます。

**Reply by BRABUS-RC** [reply]
dongabaさん、おはようございます。
テストしましたが、個別は正常に起動しませんでした。
もう少し見直しをしたいと思います。

**Reply by dongaba** [reply]
個別ダウンロードにバグがありました。
2.5.3 beta 7 で修正したつもりですのでご確認ください。
不具合が多くてすいません。

**Reply by BRABUS-RC** [reply]
dongabaさん、こんにちは。
とんでもないです。自分はPCの事も詳しくなく、分からなくて質問等をしてもいつも親切に教えて頂きありがとうございます。
また、素敵なソフトも使わさせて頂き感謝してます。
2.5.3 beta 8で快適に起動しております。
この度は、ありがとうございました。之からも宜しくお願い致します。

**Comment by BRABUS-RC** [comment]
dongabaさん、こんばんは。
今ダウンロードして気づいたのですがダウンロードさてた放送日表示に不具合があります。
以前→番組名 2022年 2022年07月29日放送 ○○○○.mp4
今日→番組名 2022年 926() ○○○○.mp4
放送日の日付表示に不具合が出てるようです。確認お願い致します。

**Reply by dongaba** [reply]
特定の動画だけで起きている問題でしょうか?
それとも、すべての動画で起きている問題でしょうか?
私の環境では再現しないので、具体的な番組名などの情報を教えてください。
また、使用しているOS、バージョンなどの情報も記載ください。

**Reply by dongaba** [reply]
Windows Powershell (Powershell Desktop)のみで問題が起きていることを確認し、v2.5.4で修正しました。

**Reply by BRABUS-RC** [reply]
こんばんは。
昨日は寝てて状況報告出来ずすみません。
今起動確認しました。正常に表示しております。
更新ありがとうございました。

**Comment by pkqah** [comment]
![1](https://user-images.githubusercontent.com/95971311/193410106-2d53f02a-f2c8-4ff0-8082-101e9d2052cd.png)

**Reply by pkqah** [reply]
![QQ截图20221001205132](https://user-images.githubusercontent.com/95971311/193410557-78bfadbb-3d51-4233-8b96-98227986536a.png)
というファイルを
![QQ截图20221001205517](https://user-images.githubusercontent.com/95971311/193410581-fa52d4eb-089e-432f-ad95-b5b65dadf85d.png)
に名前を変えることで解決しました。^^

**Reply by dongaba** [reply]
ご報告ありがとうございます。
すみません、完全に誤植ですね。
次回バージョンアップ時に直しておきます。

**Comment by BRABUS-RC** [comment]
dongabaさん、こんばんは。
お世話になってます。
ソフト更新ありがとうございます。
早速導入しましたが起動しないので調べて頂けませんか。
下記が表示され起動しません。

自動
Active code page: 65001
D:\TVerRec-2.5.6\TVerRec-2.5.6\src\tverrec_bulk_5.ps1 : 設定ファイルの読み込みに失敗しました
At line:1 char:1
+ ..\src\tverrec_bulk_5.ps1
+ ~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,tverrec_bulk_5.ps1

D:\TVerRec-2.5.6\TVerRec-2.5.6\src\delete_trash_5.ps1 : 設定ファイルの読み込みに失敗しました
At line:1 char:1
+ ..\src\delete_trash_5.ps1
+ ~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,delete_trash_5.ps1

D:\TVerRec-2.5.6\TVerRec-2.5.6\src\validate_video_5.ps1 : 設定ファイルの読み込みに失敗しました
At line:1 char:1
+ ..\src\validate_video_5.ps1
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,validate_video_5.ps1

D:\TVerRec-2.5.6\TVerRec-2.5.6\src\validate_video_5.ps1 : 設定ファイルの読み込みに失敗しました
At line:1 char:1
+ ..\src\validate_video_5.ps1
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,validate_video_5.ps1

D:\TVerRec-2.5.6\TVerRec-2.5.6\src\move_video_5.ps1 : 設定ファイルの読み込みに失敗しました
At line:1 char:1
+ ..\src\move_video_5.ps1
+ ~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,move_video_5.ps1

D:\TVerRec-2.5.6\TVerRec-2.5.6\src\delete_trash_5.ps1 : 設定ファイルの読み込みに失敗しました
At line:1 char:1
+ ..\src\delete_trash_5.ps1
+ ~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,delete_trash_5.ps1

終了するには Y と入力してください。何も入力しなければ処理を継続します。
3600秒待機します... [Y,N]?

個別
Active code page: 65001
D:\TVerRec-2.5.6\TVerRec-2.5.6\src\tverrec_single_5.ps1 : 設定ファイルの読み込みに失敗しました
At line:1 char:1
+ ..\src\tverrec_single_5.ps1
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,tverrec_single_5.ps1

Press any key to continue . . .

すみませんが宜しくお願い致します。

**Reply by dongaba** [reply]
いつもバグ報告を有難うございます。そして、たびたびの不具合ですみません。
Windows PowerShellでのみ起きる不具合を確認しましたが、Windows PowerShellに起因する不具合が多いため、そして、私が普段Windows PowerShellを使っておらず、Windows PowerShellをサポートすることに疲れたため、Windows PowerShellのサポートを終了するという形で対応しました。
PowerShell Coreを自動でインストールする機能をつけたので、PowerShell Coreをインストールしてご利用ください。

**Reply by BRABUS-RC** [reply]
おはようございます。
自動、個別起動確認しました。
ありがとうございました。

**Comment by BRABUS-RC** [comment]
dongabaさん、こんばんは。
ソフト更新お疲れ様です。
下記表示で起動できませんでした。確認お願い致します。

個別
Active code page: 65001
Write-Error: 設定ファイルの読み込みに失敗しました
Press any key to continue . . .

自動
Active code page: 65001
Write-Error: 設定ファイルの読み込みに失敗しました
Write-Error: 設定ファイルの読み込みに失敗しました
Write-Error: 設定ファイルの読み込みに失敗しました
Write-Error: 設定ファイルの読み込みに失敗しました
Write-Error: 設定ファイルの読み込みに失敗しました
Write-Error: 設定ファイルの読み込みに失敗しました
終了するには Y と入力してください。何も入力しなければ処理を継続します。
3600秒待機します... [Y,N]?

**Reply by dongaba** [reply]
すいません、凡ミスで配布する設定ファイルが壊れていました。
`system_setting.ps1`を修正して、V2.6.0としました。

**Reply by BRABUS-RC** [reply]
dongabaさん、お疲れ様です。
今、起動確認しました。個別、自動共正常に起動しております。
ありがとうございました。

**Comment by fishing-ukulele** [comment]
はじめまして。
利用させていただいております。
素晴らしいソフトありがとうございます。

tver_functions.ps1 の 620行目
$ture が $true になっているため、
youtube-dlのプロセスが増え続けてしまいます。

ご確認よろしくお願いします。

**Reply by dongaba** [reply]
丁寧に問題箇所まで調べていただきありがとうございます。
v2.6.1で修正しました。

**Reply by fishing-ukulele** [reply]
素早いご対応ありがとうございました。

**Comment by pkqah** [comment]
dongabaさん、こんばんは！いつもお世話になっております。
下記表示でダウンロードできませんでした。確認お願い致します。
![QQ图片20221019212151](https://user-images.githubusercontent.com/95971311/196703025-4dcfd9b0-c231-4967-9a9b-c3c190f950a8.png)

**Reply by dongaba** [reply]
私の環境では再現しないのでお使いの環境についてもう少し記載ください。
また、youtube-dlはyt-dlpでしょうか?それともytdl-patchedでしょうか?
一般に、このエラーメッセージはgithubへの接続に失敗した際に表示されます。

**Comment by fishing-ukulele** [comment]
いつもお世話になっております。
素晴らしいソフトありがとうございます。

「ダウンロードが ○多重に達したので一時待機します。」になって
1分間の待機中にすべての youtube-dl プロセスが終了した場合
そのまま「ダウンロードが ○多重に達したので一時待機します。」の
無限ループに入ってしまいます。

v2.6.1 では有った catch の中の
「$local:ytdlCount = 0」が無くなったからではないかと思われます。
(v2.6.2 の tver_functions.ps1 の 624行目)

ご確認よろしくお願いいたします。

**Reply by dongaba** [reply]
ご報告ありがとうございます。
IssueとしてLogしておきました。
https://github.com/dongaba/TVerRec/issues/21

**Reply by dongaba** [reply]
v2.6.3で修正しました

**Comment by BRABUS-RC** [comment]
dongabaさん、こんばんは。
ソフト更新お疲れ様です。
個別DLをすると「リストを更新できませんでした。でスキップします」と表示されます。
一度確認して頂けませんか。
自動DLは問題なく動いてます。
宜しくお願い致します。

**Reply by dongaba** [reply]
申し訳ありません。個別ダウンロードの修正を漏らしていました。
[こちら](https://github.com/dongaba/TVerRec/issues/22)に不具合記録しました。
すぐに直します。

**Reply by dongaba** [reply]
v2.6.5で修正しました。
https://github.com/dongaba/TVerRec/releases/tag/v2.6.5

まだ不具合があればご連絡ください。

**Reply by BRABUS-RC** [reply]
dongabaさん、お手数おかけしました。
問題無く起動しております。ありがとうございました。

**Comment by ghost** [comment]
お世話になっております｡
本日最新版を使用したところ､

youtube-dl(yt-dlp)の最新バージョンを特定できませんでした
ffmpegの最新バージョンを特定できませんでした　

から始まり､各キーワードに対しても

TVerから情報を取得できませんでした。スキップします Err:02

となるようです｡

**Reply by dongaba** [reply]
確認しましたが、再現しませんので、使っている環境(OS等)やTVerRecをどこからダウンロードしたのか(Git Cloneした?Releaseから?など)もう少し説明してください。
![image](https://user-images.githubusercontent.com/83079591/216827670-ca2b7983-2906-460b-b7f3-01f36496c317.png)

**Reply by ghost** [reply]
環境はWin11 powershellcore導入済み､Releaseのものを使用しています｡　

![scs](https://user-images.githubusercontent.com/103880261/216831151-a63a5b70-3e36-45f9-848b-ce1fdee8327a.png)

画像のような実行結果となり､使用できなくなりました｡　新規で settingとkeywordのみを設定しましたが動きません｡　

なお原因がわかったのですが､今までできてたVPNで接続ができなくなったようです｡　通常接続だと動くようです｡

**Reply by dongaba** [reply]
見る限りネットワーク接続に問題があるようですね。
VPNを利用されているとのことですが、もし日本国外からご利用でTVerのためにVPN接続しているのであれば、TVerRecに関してはVPN接続しなくてもダウンロード可能です。ご参考までに。

**Reply by dongaba** [reply]
なお、4つ出ているエラーの1つ目については、dbフォルダ配下にtver.csvとhistory.csvの双方があることが原因と思われます。
v2.6.5以前からのバージョンアップであれば、tver.csvをhistory.cvsに名称変更しようとしているのですが、既にhistory.csvが存在するのでエラーとなっていると思われます。
不要な方のファイルを削除すればエラーはなくなると思われます。
ほかの3つのエラーはネットワーク接続がないことが原因と思われます。

**Reply by ghost** [reply]
VPNに関しては相性があったようで､VPNの設定変更で対処できました｡　
以前まではtver.csvとhistory..csvが共存した状態であってもエラーなく動いていたのでなにかサイト側で使用が変わったのかと勘違いしました｡　大変お手数おかけしました､いつもありがとうございます！

**Comment by fishing-ukulele** [comment]
いつもありがとうございます。

v2.7.1で
$script:simplifiedValidation が $true の時
エラーになってしまい
一旦ダウンロードしたファイルがチェックで削除されてしまいます。

tver_functions.ps1 の 2305行目と2307行目が
それぞれ " と ' になっているためのようです。

ご確認よろしくお願いします。

**Reply by dongaba** [reply]
不具合報告をありがとうございます。
v2.7.2で修正しました。
文法エラーにならないバグだったため気がついておらず、不具合箇所の特定までしていただき助かりました。

**Reply by fishing-ukulele** [reply]
ご対応ありがとうございました。
